### PR TITLE
[Parallel Run Step] Support run settings binding to pipeline literal input.

### PR DESCRIPTION
**Pull request by bupt-wenxiaole**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by elliotzh** [comment]
@bupt-wenxiaole Hi Xiaole, could you please share some background what do you want to do with this pull request? data binding expression has already been supported for Prallel node by dynamically schema update, so I wonder in what case you failed to use data-binding expression

**Review by elliotzh — CHANGES_REQUESTED** [review]
@bupt-wenxiaole Hi Xiaole, could you please share some background what do you want to do with this pull request? data binding expression has already been supported for Prallel node by dynamically schema update, so I wonder in what case you failed to use data-binding expression

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py [hunk 1]
```diff
@@ -266,9 +266,9 @@ def _attr_type_map(cls) -> dict:
             "resources": (dict, JobResourceConfiguration),
             "task": (dict, ParallelTask),
             "logging_level": str,
-            "max_concurrency_per_instance": int,
-            "error_threshold": int,
-            "mini_batch_error_threshold": int,
+            "max_concurrency_per_instance": (str, int),
```
Comment (elliotzh) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py:269]: this should be the only necessary change. Please add test and check if you can revert changes other than attr_map

File: sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py [hunk 2]
```diff
@@ -1711,6 +1712,124 @@ def parallel_in_pipeline(job_data_path):
         assert_job_input_output_types(pipeline_job)
         assert pipeline_job.settings.default_compute == "cpu-cluster"
 
+    def test_parallel_run_function_run_settings_bind_to_liter_input(self, client: MLClient):
+        # command job with dict distribution
+        environment = "AzureML-sklearn-1.0-ubuntu20.04-py38-cpu:33"
```
Comment (elliotzh) [inline sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py:1717]: maybe you can extract the shared part comparing to `test_parallel_run_function` as a function like `build_pipeline_with_parallel_run`

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py [hunk 3]
```diff
@@ -357,8 +357,7 @@ def _from_rest_object_to_init_params(cls, obj: dict) -> Dict:
                 obj["task"].environment = task_env[len(ARM_ID_PREFIX) :]
 
         if "resources" in obj and obj["resources"]:
-            resources = RestJobResourceConfiguration.from_dict(obj["resources"])
-            obj["resources"] = JobResourceConfiguration._from_rest_object(resources)
+            obj["resources"] = JobResourceConfiguration._from_dict(obj["resources"])
```
Comment (elliotzh) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py:360]: why need a new function here?

File: sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py (hunk 2 above)
Comment (bupt-wenxiaole) [reply sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py:1717]: fixed.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py (hunk 1 above)
Comment (bupt-wenxiaole) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py:267]: reverted the schema change

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py (hunk 3 above)
Comment (bupt-wenxiaole) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py:358]: the RestJobResourceConfiguration has attribute type checker, the node_count should be a data binding string instead of int type here, so here I changed to load JobResourceConfiguration from dict directly.

File: sdk/ml/azure-ai-ml/azure/ai/ml/_scope_dependent_operations.py [hunk 4]
```diff
@@ -21,7 +21,6 @@ class OperationConfig(object):
     """
 
     def __init__(self, show_progress: bool, enable_telemetry: bool) -> None:
-
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_scope_dependent_operations.py:24]: seems unrelated change?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_artifacts/_artifact_utilities.py [hunk 5]
```diff
@@ -317,7 +317,6 @@ def _upload_and_generate_remote_uri(
     datastore_name: str = WORKSPACE_BLOB_STORE,
     show_progress: bool = True,
 ) -> str:
-
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_artifacts/_artifact_utilities.py:320]: seems unrelated change?  revert?

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/_scope_dependent_operations.py (hunk 4 above)
Comment (bupt-wenxiaole) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_scope_dependent_operations.py]: reverted.

**Review by elliotzh — APPROVED** [review]
